python/pyarrow/lib.pxd

```diff
@@ -469,15 +469,19 @@ cdef class ChunkedArray(_PandasConvertible):
     cdef getitem(self, int64_t i)
 
 
-cdef class Table(_PandasConvertible):
+cdef class _Table(_PandasConvertible):
```

Comment:
Do we maybe want to use "Base" in the name (to make it clearer that it is a shared base class?) `_BaseTable` would be fine (or `_BaseTabular`, if that's a better word to describe the commonality between RecordBatch and Table)

Reply:
How about `_Tabular`? It's an adjective so `class RecordBatch(_Tabular)` and `class Table(_Tabular)` sound very descriptive IMO. My initial thought is that adding `Base` here is redundant, but let me know if you think otherwise.

Reply:
"tabular" sounds good!